### DISCUSSION TITLE
6.2.1 (Grid Effects alpha, stability fixes)

### DISCUSSION THREAD

**Discussion by T-Troll** [T-Troll/alienfx-tools#155] [Announcements]
### Changes in this release:

`AlienFX-GUI`:
- New: Grid effect implementation:
  - All "Grid types" implemented (but "Gradient" is not in final form)
  - "Keyboard" trigger implemented (see updated [manual](https://github.com/T-Troll/alienfx-tools/wiki/Grid-Effects) for details!)
  - "Circle" and "Zone lights only" check box implemented
  - "Speed" slider re-worked - now it can be set from -80 to +80. Zero is one cell per tact (100ms), negative value is divide (so -5 means 1/5 cell per tact), positive is multiplier (so 4 means 4 cells per tact).
  - Physical light updates optimized, no more full-zone set at each update.
- Profile switch by key is optimized

Fan-control GUI apps:
- Added some checks to prevent potential exceptions (some m15R6's unstable here). This also decrease CPU load if app/tab opened.

Known limitations:
- Grid effect settings still not saved between sessions. I will change some data structures after final touches, so now it's not final.

<hr /><em>This discussion was created from the release <a href='https://github.com/T-Troll/alienfx-tools/releases/tag/6.2.1'>6.2.1 (Grid Effects alpha, stability fixes)</a>.</em>

**Comment by MarlosLucas** [comment]
Hey T-Troll! How are you doing?

Regarding the last update (hotfix 6.2.1.1), should I restart the windows to make effect? Just to know because after to install it, the same issue is still alive. I am running a lot of models stuffs and I cannot do that right now. But please let me know if a restart is requested.

Cheers and good job!

**Reply by T-Troll** [reply]
No, it doesn't require reboot.
What error message do you have trying to enable fan control? It's some of them for different situations!

**Reply by MarlosLucas** [reply]
![image](https://user-images.githubusercontent.com/37800685/175793662-7ac0f9ea-5ff1-4fde-a358-ec236e430cd8.png)

**Reply by T-Troll** [reply]
I see. This is different issue, i fix "Hardware not detected" one.
This one related to weird Windows 11 "security" they change every update, i'm a bit tired to fight this. The topic i mention in previous discussion has some sidewalks to overtake it using KDU, but no universal way.

I'm starting to think it will be a good idea to drop official Windows 11 support completely.

**Reply by MarlosLucas** [reply]
I`ll keep trying the resolution of this %*@#&. 
Regarding the new complete support for W11.. Man, I would love this. But I can understand you. Take it your time!

Cheers!

**Reply by T-Troll** [reply]
Hmm... Things become even more interesting. My testers have identical hardware (m15R6), identical OS (Windows 11), but it starts at one system and didn't start at the other! Seems like some weird setting...

**Reply by MarlosLucas** [reply]
Hey, T-Troll

I have news about my issue. I could note that my alienfx runs perfectly after turn-off and turn-on the PC. Restarting, the PC the software already starts affected.

**Reply by T-Troll** [reply]
Do you have "Fast startup" enabled at Windows Power Plan page?
If yes, disable it - and check.

**Reply by MarlosLucas** [reply]
Done! In few minutes, I will try to restart and bring the results here!

thanks, t-troll

**Reply by MarlosLucas** [reply]
Hey, T-Troll

I think thats solved!

After some restarts, looks stable!

**Reply by T-Troll** [reply]
Aye. I also add potential fix (but not sure if it's enough) into 6.2.3 driver.
I update readme about this.
